src/WikiService.Business/Commands/Rubric/Edit/EditRubricHandler.cs

```diff
@@ -27,17 +28,56 @@ private Task<DbRubric> GetAsync(Guid rubricId, CancellationToken ct)
       return _provider.Rubrics.FirstOrDefaultAsync(x => x.Id == rubricId, ct);
     }
 
+    private async Task ChangeActivationAsync(Guid rubricId, bool isActivate)
+    {
+      List<DbRubric> rubrics = await _provider.Rubrics.Where(rubric => rubric.ParentId == rubricId).ToListAsync();
+
+      foreach (DbRubric rubric in rubrics)
+      {
+        List<DbArticle> articles = await _provider.Articles.Where(article => article.RubricId == rubric.Id).ToListAsync();
+        foreach (DbArticle article in articles)
+        {
+          article.IsActive = isActivate;
+        }
+
+        rubric.IsActive = isActivate;
+
+        await ChangeActivationAsync(rubric.Id, isActivate);
+      }
+    }
+
     private async Task<bool> EditAsync(DbRubric dbRubric, JsonPatchDocument<DbRubric> request)
     {
       if (dbRubric is null || request is null)
       {
         return false;
       }
 
-      request.ApplyTo(dbRubric);
+      Guid? rubricId = dbRubric.ParentId;
+      bool isAtive = dbRubric.IsActive;
+      request.ApplyTo(dbRubric);
+
+      if ((rubricId != dbRubric.ParentId || isAtive != dbRubric.IsActive)
+        && dbRubric.IsActive
+        && !(await _provider.Rubrics.FirstOrDefaultAsync(rubric => rubric.Id == dbRubric.ParentId)).IsActive)
+      {
+        throw new BadRequestException("Parent rubric is not active.");

```

Comment:
may be 'Active sub-rubric can't be in archive rubric'

Reply:
done